filter_k8s_servicetypes.go

```diff
@@ -0,0 +1,32 @@
+package aiven
+
+var _ Filter = (*k8sServiceTypesFilter)(nil)
+
+// k8sSupportedServiceTypes is a map of service types supported in Kubernetes operator.
+var k8sSupportedServiceTypes = map[string]struct{}{
+	"clickhouse":    {},
+	"kafka":         {},
+	"kafka_connect": {},
+	"opensearch":    {},
+	"pg":            {},
+	"redis":         {},
+}
+
+// k8sServiceTypesFilter is a filter implementation that filters out service types not supported in Kubernetes operator.
+type k8sServiceTypesFilter struct{}
+
+// Filter filters the given elements.
+func (f *k8sServiceTypesFilter) Filter(fe FilterableElements) (FiltrationResult, error) {
```

Comment:
Since these filters are only applied for one function and one function only where we predicatively have `map[string]ServiceType` in use, should we change it here as well? The same with the `FiltrationResult`

I see it more like: `Filter (fe map[string]ServiceType) (map[string]ServiceType, error)` or even `Filter (fe map[string]ServiceType) error`

Reply:
Empty interface type casting here is first unnecessary and may lead to unnecessary errors. Since we can avoid this by going away from empty interfaces, I think changing is a good idea. If we see the need for it in the future, we can change it down the road.

Reply:
in the updated version there is a necessary contract binding for the `Filter` interface, thus plugging in concrete types will cause it to be violated

Reply:
this is no longer relevant